lib/chef/provider/file.rb

```diff
@@ -334,7 +334,7 @@ def tempfile_checksum
       end
 
       def do_validate_content
-        if new_resource.checksum && tempfile && ( new_resource.checksum.downcase != tempfile_checksum )
```

Comment:
are we coercing this in file as well?

Reply:
I did that in this PR https://github.com/chef/chef/pull/9886/commits/60f2c4a8dfed08656def3abef6f6a11fa2ef8f2f